pkg/sql/compile/compile.go

```diff
@@ -856,6 +856,12 @@ func (c *Compile) compilePlanScope(ctx context.Context, step int32, curNodeIdx i
 	case plan.Node_AGG:
 		curr := c.anal.curr
 		c.setAnalyzeCurrent(nil, int(n.Children[0]))
+
+		child := ns[n.Children[0]]
+		if child.NodeType == plan.Node_TABLE_SCAN {
```

Comment:
Modifications to plan MUST NOT be outside planner/optimizer.

You should put it, and the "if" logic under pkg/sql/plan. The final output of planner/optimizer should be clean and all-inclusive. Pipeline just loyally executes what plan tells to do, no more and no less.